src/app/notifications/notifications.component.ts

```diff
@@ -54,6 +56,13 @@ export class NotificationsComponent implements OnInit, AfterViewInit {
     }, (err) => console.log(err.error.reason));
   }
 
+  markRead(notification) {
```

Comment:
this function is not required as it is already done by readNotification function. You just need a way to call readNotification without triggering link